### PR TITLE
Refactor nf-test CI and test and other pre-release fixes

### PR DESCRIPTION
**Pull request by drpatelh**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -96,8 +97,12 @@ jobs:
           sudo apt install icdiff
 
       - name: Run nf-test
+        uses: pcolby/tap-summary@v1
+        with:
+          path: >-
+            test.tap
```
Comment (maxulysse) [inline .github/workflows/ci.yml:103]: should this be indented?

Comment (drpatelh) [reply .github/workflows/ci.yml:103]: The `with` statement in the section above is indented 🤷🏽

Comment (maxulysse) [reply .github/workflows/ci.yml:103]: yeah, I need to brush up with GHA

**Review by maxulysse — APPROVED** [review]
Looking good, just a couple of failing tests to fix and we're good to go

**Comment by drpatelh** [comment]
Merging in for now since we all tests are now passing. We can push further updates in future PRs if required.

File: tests/nextflow.config [hunk 2]
```diff
@@ -0,0 +1,18 @@
+params {
+    // Base directory for nf-core/modules test data
+    modules_test_data_base = 'https://raw.githubusercontent.com/nf-core/test-datasets/3e79d4b5fbed8c6ef7546d50696b343ab9a95fbe/'
+
+    // Base directory for pipeline test data
+    pipelines_test_data_base = 'https://raw.githubusercontent.com/nf-core/test-datasets/2732b911c57e607fa7aea5ba0c3d91b25bafb662/'
```
Comment (grst) [inline tests/nextflow.config:6]: I'm wondering if we could eventually get rid of this distinction and move to a single branch/bucket with test data that is used across modules and all pipelines. The "modules" branch is already pretty broad in scope and should cover a lot of the various pipelines' test datasets already.

**Pull request merged**